ctru-rs/src/applets/swkbd.rs

```diff
@@ -311,6 +368,71 @@ impl SoftwareKeyboard {
         self.state.filter_flags = filters.bits();
     }
 
+    /// Configure a custom filtering function to validate the input.
+    ///
+    /// The callback function will return a callback result and the error message to display when the input is invalid.
+    ///
+    /// # Notes
+    ///
+    /// The filter callback will work only for the next time the keyboard is used. After using it once, it must be set again.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # fn main() {
+    /// #
+    /// use ctru::applets::swkbd::{SoftwareKeyboard, CallbackResult};
+    /// let mut keyboard = SoftwareKeyboard::default();
+    ///
+    /// keyboard.set_filter_callback(|text| {
+    ///     if text.contains("boo") {
+    ///         println!("Ah, you scared me!");
+    ///     }
+    ///
+    ///     (CallbackResult::Ok, None)
+    /// });
+    /// #
+    /// # }
+    pub fn set_filter_callback<F>(&mut self, callback: F)
```

Comment:
Something else just occurred to me: should we have `Option<F>` here so that users can also clear the callback if it was previously set? They could of course just pass a trivial callback like `|_| Callbackresult::Ok` but not sure if there's any reason to set the callback to `null` (via `None`) explicitly.

Reply:
That's pretty much necessary if we decide to store the closure within the `SoftwareKeyboard` configuration.